### PR TITLE
inits logs collection blog post

### PR DESCRIPTION
**Pull request by mhausenblas**
[Preview](https://deploy-preview-3557--opentelemetry.netlify.app/blog/2023/logs-collection/)

~The site currently doesn't build due to some Hugo/Docsy fun I am not able to fix myself (I suspect due to the non-standard sub-directory structure in the blog post itself I have, with things like `yoda/` that contains no Markdown files.~

Fixed by running `npm install`, kudos to @svrnm for the help!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@ tmp/
 
 # macOS
 .DS_Store
+.DS-Store
```
Comment (chalin) [inline .gitignore:15]: To my knowlegde, `.DS-Store` isn't a legit macOS name, so I've vote against adding this here:

```suggestion
```

Besides, the files got committed anyhow! 😄 
They need to be removed from the commit.

**Review by chalin — CHANGES_REQUESTED** [review]
Quick comment before I take a deeper dive.

File: .gitignore (hunk 1 above)
Comment (mhausenblas) [reply .gitignore:15]: For the record: I was unable to resolve this issue, that is, [figure out why my macOS version does that](https://x.com/jkowall/status/1726428762965721302?s=20), so, accepting it tho I still don't know what's going on ha ha

File: content/en/blog/2023/logs-collection/index.md [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+## Expert Grogu level
+
+Over time, Baby Grogu has learned about observability and telemetry
+specifically. They have advanced to Expert Grogu level. How? Glad you asked, let
+me show you.
+
+First, change into the [expert-grogu/][repo-expert-grogu] directory.
+
+In this scenario Expert Grogu is logging into a file (in JSON format), from
+their Python app. Then, they are using the OTel collector to read that very log
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:121]:
```suggestion
their Python app. Then, they are using the OpenTelemetry Collector to read that very log
```

File: content/en/blog/2023/logs-collection/index.md [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+file, parse the log records using the [filelog receiver][filelog] in the OTel
+collector and finally output it to `stdout` using the [debug exporter][debug].
+Makes sense? Let's see it in action …
+
+Overall, we have the following setup:
+
+```plain
+( python main.py ) --> exgru.log --> ( OTel collector ) --> stdout
+```
```
[the remark this answers is not in the document]
Comment (svrnm) [reply content/en/blog/2023/logs-collection/index.md:130]:
```suggestion
```plain
( python main.py ) --> exgru.log --> ( OTel Collector ) --> stdout
```
```

File: content/en/blog/2023/logs-collection/index.md [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+
+```docker
+FROM python:3.11
+WORKDIR /usr/src/app
+COPY requirements.txt requirements.txt
+RUN pip3 install --no-cache-dir -r requirements.txt
+COPY . .
+```
+
+With the following OTel collector config (visualize via
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:170]:
```suggestion
With the following OTel Collector config (visualize via
```

File: content/en/blog/2023/logs-collection/index.md [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+
+## Yoda level
+
+Now we're switching gears and look over Yoda's shoulders, a Telemetry Master.
+
+First, change into the [yoda/][repo-yoda] directory.
+
+In this scenario we see Yoda using the OTel logs bridge API in the Python app to
+directly ingest logs, in [OpenTelemetry Protocol][otlp] (OTLP) format, into the
+OTel collector. This is both faster and more reliable than first logging to a
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:302]:
```suggestion
OTel Collector. This is both faster and more reliable than first logging to a
```

File: content/en/blog/2023/logs-collection/index.md [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+  debug:
+    verbosity: detailed
+service:
+  pipelines:
+    logs:
+      receivers: [otlp]
+      exporters: [debug]
+```
+
+Now run Yoda's setup with `docker-compose -f docker-compose.yaml` and you should
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:329]:
```suggestion
Now run Yoda's setup with `docker compose -f docker-compose.yaml` and you should
```

`docker-compose` is deprecated use `docker compose`

File: content/en/blog/2023/logs-collection/index.md [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+services:
+  baby-grogu:
+    build: .
+    command: python main.py 3
+    volumes:
+      - .:/usr/src/app
+```
+
+At this point you can enjoy Baby Grogu's efforts by running it with
+`docker-compose -f docker-compose.yaml` and you should see an output akin to
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:98]:
```suggestion
`docker compose -f docker-compose.yaml` and you should see an output akin to
```

`docker-compose` is deprecated use `docker compose`

File: content/en/blog/2023/logs-collection/index.md [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+    ports:
+      - '4317:4317'
+  baby-grogu:
+    build: .
+    command: python main.py 10
+    volumes:
+      - .:/usr/src/app
+```
+
+Which you can run it with `docker-compose -f docker-compose.yaml` and you should
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:214]:
```suggestion
Which you can run it with `docker-compose -f docker-compose.yaml` and you should
```

Use `docker compose`, `docker-compose` is deprecated

File: content/en/blog/2023/logs-collection/index.md [hunk 9]
```diff
@@ -0,0 +1,466 @@
...
+yoda-collector-1   | Flags: 0
+yoda-collector-1   |    {"kind": "exporter", "data_type": "logs", "name": "logging"}
+yoda-baby-grogu-1  | =`;*'+.|,+?):(*-<}~}
+```
+
+Fun, hu? You can play around with Yoda's source code to add more contextual
+information and add processors to manipulate the log records as they pass the
+collector, now.
+
+May The Telemetry be with you, young Padawan!
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:401]:
```suggestion
May _The Telemetry_ be with you, young Padawan!
```

File: content/en/blog/2023/logs-collection/index.md [hunk 10]
```diff
@@ -0,0 +1,466 @@
...
+
+## What's next?
+
+Now that you're familiar with The Telemetry and its good practices, you could
+extend Yoda's code to do the following:
+
+1. Add more context. For example, try to use OTel resource attributes and the
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:408]:
```suggestion
1. Add more context. For example, try to use [OTel resource attributes](/docs/concepts/resources/) and the
```

File: content/en/blog/2023/logs-collection/index.md [hunk 11]
```diff
@@ -0,0 +1,466 @@
...
+   semantic conventions to make the context of the execution more explicit.
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:409]:
```suggestion
   [semantic conventions](/docs/concepts/semantic-conventions/) to make the context of the execution more explicit.
```

File: content/en/blog/2023/logs-collection/index.md [hunk 12]
```diff
@@ -0,0 +1,466 @@
...
+1. Enrich the logs in the OTel collector or filter certain severity levels,
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:410]:
```suggestion
1. Enrich the logs in the OTel Collector or filter certain severity levels,
```

File: content/en/blog/2023/logs-collection/index.md [hunk 13]
```diff
@@ -0,0 +1,466 @@
...
+   using processors such as the transform or attributes processors.
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:411]:
```suggestion
   using processors such as the [transform or attributes processors](/docs/collector/transforming-telemetry/).
```

File: content/en/blog/2023/logs-collection/index.md [hunk 14]
```diff
@@ -0,0 +1,466 @@
...
+1. Add tracing support by emitting spans, where it makes sense.
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:412]:
```suggestion
1. Add [tracing](/docs/concepts/signals/traces/) support by emitting spans, where it makes sense.
```

File: content/en/blog/2023/logs-collection/index.md [hunk 15]
```diff
@@ -0,0 +1,466 @@
...
+1. Add an o11y backend such as OpenSearch (along with [Data
+   Prepper][dataprepper]) to the setup, allowing to ingest spans and logs in
+   OTLP format.
+1. Once you have traces and logs ingested in a backend, try to correlate these
+   two telemetry signal types in the backend along with a frontend such as
+   Grafana.
+1. Use auto-instrumentation to further enrich telemetry.
```
Comment (svrnm) [inline content/en/blog/2023/logs-collection/index.md:419]:
```suggestion
1. Use [Automatic Instrumentation](/docs/concepts/instrumentation/automatic/) to further enrich telemetry.
```

**Review by svrnm — CHANGES_REQUESTED** [review]

File: content/en/blog/2023/logs-collection/index.md [hunk 16]
```diff
@@ -0,0 +1,466 @@
...
+demonstration programming language, however note that at time of writing the
+logs support there is still early days so things might need some updating.
+
+We will show the evolution from using print statements for logging (_Baby Grogu_
+level) to logging to a file along with the OTel collector (_Expert Grogu_ level)
+to using the OTel logs bridge API to directly ingest OTLP (_Yoda_ level) into
+the collector.
+
+If you want to follow along, you need Docker installed and first off, go ahead
+and `git clone https://github.com/open-telemetry/opentelemetry.io.git` and
```
[the remark this answers is not in the document]
Comment (mhausenblas) [reply content/en/blog/2023/logs-collection/index.md]: Done

**Comment by mhausenblas** [comment]
Thanks for your review @svrnm and I addressed all of your comments including moving the code out and upgrading to Mermaid graphs.

File: content/en/blog/2023/logs-collection/index.md [hunk 17]
```diff
@@ -0,0 +1,481 @@
...
+yoda-collector-1   | SeverityNumber: Info(9)
+yoda-collector-1   | Body: Str(Practicing The Telemetry completed: True)
+yoda-collector-1   | Trace ID:
+yoda-collector-1   | Span ID:
+yoda-collector-1   | Flags: 0
+yoda-collector-1   |    {"kind": "exporter", "data_type": "logs", "name": "logging"}
+yoda-baby-grogu-1  | =`;*'+.|,+?):(*-<}~}
+```
+
+Fun, hu? You can play around with Yoda's source code to add more contextual
```
Comment (cartermp) [inline content/en/blog/2023/logs-collection/index.md:404]:
```suggestion
Fun, huh? You can play around with Yoda's source code to add more contextual
```

**Review by cartermp — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Pull request merged**